### ISSUE TITLE
源码安装错误

### ISSUE THREAD

**Issue by Stu-Yang** [secretflow/secretflow#245]
### Issue Type

Build/Install

### Source

source

### Secretflow Version

secretflow v0.7.11b3

### OS Platform and Distribution

Linux Ubuntu 20.04

### Python version

Python 3.8.15

### Bazel version

bazel 5.1.1

### GCC/Compiler version

gcc 9.4.0

### What happend and What you expected to happen.

```shell
当我在按照[INSTALLATION.md](https://github.com/secretflow/secretflow/blob/main/INSTALLATION.md)中的[Option 3: from source](https://github.com/secretflow/secretflow/blob/main/INSTALLATION.md#option-3-from-source)进行安装时发送了错误。
```

### Reproduction code to reproduce the issue.

```shell
>>> git clone https://github.com/secretflow/secretflow.git
>>> cd secretflow

>>> conda create -n secretflow python=3.8
>>> conda activate secretflow

>>> python setup.py bdist_wheel       # 在这一步出现了错误
Requirements:  ['numpy>=1.22.0', 'protobuf==3.19.4', 'jax==0.3.17', 'jaxlib==0.3.15', 'flax==0.6.0', '', 'tensorflow_datasets', 'absl-py>=1.1.0', 'cloudpickle>=2.0.0', 'networkx~=2.8', '', 'secretflow-ray==2.0.0.dev1', 'tensorflow==2.10.0; platform_machine != "arm64"', 'tensorflow-macos==2.10.0; platform_machine == "arm64" and platform_system == "Darwin"', '', 'pandas', 'pyarrow', 'scikit-learn==1.1.3', '', 'torch==1.10.1', 'torchvision==0.11.2', 'torchmetrics==0.9.2', '', 's3fs==2022.1.0', 'xgboost==1.5.2', 'validator.py==1.3.0', 'tqdm', 'loguru', 'statsmodels==0.13.2', '', 'matplotlib', 'multiprocess', 'cachetools', '', '# for tests', 'psutil', 'coverage', 'unittest-xml-reporting', 'spu==0.3.0', 'sf-heu==0.3.0b0']
/root/miniconda3/envs/secretflow/lib/python3.8/site-packages/setuptools/dist.py:771: UserWarning: Usage of dash-separated 'index-url' will not be supported in future versions. Please use the underscore name 'index_url' instead
  warnings.warn(
running bdist_wheel
running build
running build_py
running build_ext
bazel build //secretflow_lib/binding:_lib.so --symlink_prefix=build/temp.linux-x86_64-cpython-38/bazel- --compilation_mode=opt
INFO: Analyzed target //secretflow_lib/binding:_lib.so (0 packages loaded, 0 targets configured).
INFO: Found 1 target...
ERROR: /root/.cache/bazel/_bazel_root/ab0fc7d3e6dc917b73f2787ab1aea828/external/com_github_google_cpu_features/BUILD.bazel:10:20: Foreign Cc - CMake: Building cpu_features failed: (Exit 127): bash failed: error executing command /bin/bash -c bazel-out/k8-opt/bin/external/com_github_google_cpu_features/cpu_features_foreign_cc/wrapper_build_script.sh

Use --sandbox_debug to see verbose messages from the sandbox
rules_foreign_cc: Build failed!
rules_foreign_cc: Keeping temp build directory and dependencies directory for debug.
rules_foreign_cc: Please note that the directories inside a sandbox are still cleaned unless you specify --sandbox_debug Bazel command line flag.
rules_foreign_cc: Printing build logs:
_____ BEGIN BUILD LOGS _____

Bazel external C/C++ Rules. Building library cpu_features

Environment:______________
BUILD_SCRIPT=bazel-out/k8-opt/bin/external/com_github_google_cpu_features/cpu_features_foreign_cc/build_script.sh
EXT_BUILD_ROOT=/root/.cache/bazel/_bazel_root/ab0fc7d3e6dc917b73f2787ab1aea828/sandbox/linux-sandbox/233/execroot/secretflow
BUILD_LOG=bazel-out/k8-opt/bin/external/com_github_google_cpu_features/cpu_features_foreign_cc/CMake.log
PWD=/root/.cache/bazel/_bazel_root/ab0fc7d3e6dc917b73f2787ab1aea828/sandbox/linux-sandbox/233/execroot/secretflow
BUILD_WRAPPER_SCRIPT=bazel-out/k8-opt/bin/external/com_github_google_cpu_features/cpu_features_foreign_cc/wrapper_build_script.sh
BAZEL_LINKOPTS=-static-libstdc++:-static-libgcc
TMPDIR=/tmp
EXT_BUILD_DEPS=/root/.cache/bazel/_bazel_root/ab0fc7d3e6dc917b73f2787ab1aea828/sandbox/linux-sandbox/233/execroot/secretflow/bazel-out/k8-opt/bin/external/com_github_google_cpu_features/cpu_features.ext_build_deps
BAZEL_LINKLIBS=-l%:libstdc++.a:-l%:libgcc.a
BUILD_TMPDIR=/root/.cache/bazel/_bazel_root/ab0fc7d3e6dc917b73f2787ab1aea828/sandbox/linux-sandbox/233/execroot/secretflow/bazel-out/k8-opt/bin/external/com_github_google_cpu_features/cpu_features.build_tmpdir
SHLVL=2
INSTALLDIR=/root/.cache/bazel/_bazel_root/ab0fc7d3e6dc917b73f2787ab1aea828/sandbox/linux-sandbox/233/execroot/secretflow/bazel-out/k8-opt/bin/external/com_github_google_cpu_features/cpu_features
PATH=/root/.cache/bazel/_bazel_root/ab0fc7d3e6dc917b73f2787ab1aea828/sandbox/linux-sandbox/233/execroot/secretflow:/root/miniconda3/envs/secretflow/bin:/root/miniconda3/condabin:/root/.vscode-server/bin/8dfae7a5cd50421d10cd99cb873990460525a898/bin/remote-cli:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/root/bin
_=/usr/bin/env
__________________________
+ cmake -DCMAKE_INSTALL_LIBDIR=lib -DCMAKE_POSITION_INDEPENDENT_CODE=ON -DBUILD_TESTING=OFF -DCMAKE_TOOLCHAIN_FILE=/root/.cache/bazel/_bazel_root/ab0fc7d3e6dc917b73f2787ab1aea828/sandbox/linux-sandbox/233/execroot/secretflow/bazel-out/k8-opt/bin/external/com_github_google_cpu_features/cpu_features.build_tmpdir/crosstool_bazel.cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=/root/.cache/bazel/_bazel_root/ab0fc7d3e6dc917b73f2787ab1aea828/sandbox/linux-sandbox/233/execroot/secretflow/bazel-out/k8-opt/bin/external/com_github_google_cpu_features/cpu_features -DCMAKE_PREFIX_PATH=/root/.cache/bazel/_bazel_root/ab0fc7d3e6dc917b73f2787ab1aea828/sandbox/linux-sandbox/233/execroot/secretflow/bazel-out/k8-opt/bin/external/com_github_google_cpu_features/cpu_features.ext_build_deps -DCMAKE_RANLIB= -DCMAKE_MAKE_PROGRAM=ninja -G Ninja /root/.cache/bazel/_bazel_root/ab0fc7d3e6dc917b73f2787ab1aea828/sandbox/linux-sandbox/233/execroot/secretflow/external/com_github_google_cpu_features
bazel-out/k8-opt/bin/external/com_github_google_cpu_features/cpu_features_foreign_cc/build_script.sh: line 72: cmake: command not found
_____ END BUILD LOGS _____
rules_foreign_cc: Build wrapper script location: bazel-out/k8-opt/bin/external/com_github_google_cpu_features/cpu_features_foreign_cc/wrapper_build_script.sh
rules_foreign_cc: Build script location: bazel-out/k8-opt/bin/external/com_github_google_cpu_features/cpu_features_foreign_cc/build_script.sh
rules_foreign_cc: Build log location: bazel-out/k8-opt/bin/external/com_github_google_cpu_features/cpu_features_foreign_cc/CMake.log

Target //secretflow_lib/binding:_lib.so failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 0.458s, Critical Path: 0.21s
INFO: 33 processes: 33 internal.
FAILED: Build did NOT complete successfully
error: command '/root/bin/bazel' failed with exit code 1

其他信息：
```shell
>>> pip list
Package                      Version
---------------------------- ----------
absl-py                      1.3.0
aiobotocore                  2.1.2
aiohttp                      3.8.3
aioitertools                 0.11.0
aiosignal                    1.3.1
astunparse                   1.6.3
async-timeout                4.0.2
attrs                        22.2.0
botocore                     1.23.24
cachetools                   5.2.0
certifi                      2022.12.7
charset-normalizer           2.1.1
chex                         0.1.5
click                        8.0.4
cloudpickle                  2.2.0
colorama                     0.4.6
commonmark                   0.9.1
contourpy                    1.0.6
coverage                     7.0.1
cycler                       0.11.0
dill                         0.3.6
distlib                      0.3.6
dm-tree                      0.1.8
etils                        0.9.0
filelock                     3.9.0
flatbuffers                  22.12.6
flax                         0.6.0
fonttools                    4.38.0
frozenlist                   1.3.3
fsspec                       2022.1.0
gast                         0.4.0
google-auth                  2.15.0
google-auth-oauthlib         0.4.6
google-pasta                 0.2.0
googleapis-common-protos     1.56.4
grpcio                       1.51.1
h5py                         3.7.0
idna                         3.4
importlib-metadata           5.2.0
importlib-resources          5.10.2
jax                          0.3.17
jaxlib                       0.3.15
jmespath                     0.10.0
joblib                       1.2.0
jsonschema                   4.17.3
keras                        2.10.0
Keras-Preprocessing          1.1.2
kiwisolver                   1.4.4
libclang                     14.0.6
loguru                       0.6.0
lxml                         4.9.2
Markdown                     3.4.1
MarkupSafe                   2.1.1
matplotlib                   3.6.2
msgpack                      1.0.4
multidict                    6.0.4
multiprocess                 0.70.14
networkx                     2.8.8
numpy                        1.24.1
oauthlib                     3.2.2
opt-einsum                   3.3.0
optax                        0.1.4
packaging                    22.0
pandas                       1.5.2
patsy                        0.5.3
Pillow                       9.3.0
pip                          22.3.1
pkgutil_resolve_name         1.3.10
platformdirs                 2.6.2
promise                      2.3
protobuf                     3.19.4
psutil                       5.9.4
pyarrow                      10.0.1
pyasn1                       0.4.8
pyasn1-modules               0.2.8
Pygments                     2.13.0
pyparsing                    3.0.9
pyrsistent                   0.19.3
python-dateutil              2.8.2
pytz                         2022.7
PyYAML                       6.0
requests                     2.28.1
requests-oauthlib            1.3.1
rich                         11.2.0
rsa                          4.9
s3fs                         2022.1.0
scikit-learn                 1.1.3
scipy                        1.9.3
secretflow-ray               2.0.0.dev1
setuptools                   65.5.0
six                          1.16.0
statsmodels                  0.13.2
tensorboard                  2.10.1
tensorboard-data-server      0.6.1
tensorboard-plugin-wit       1.8.1
tensorflow                   2.10.0
tensorflow-datasets          4.8.0
tensorflow-estimator         2.10.0
tensorflow-io-gcs-filesystem 0.29.0
tensorflow-metadata          1.12.0
termcolor                    2.1.1
threadpoolctl                3.1.0
toml                         0.10.2
toolz                        0.12.0
torch                        1.10.1
torchmetrics                 0.9.2
torchvision                  0.11.2
tqdm                         4.64.1
typing_extensions            4.4.0
unittest-xml-reporting       3.2.0
urllib3                      1.26.13
validator.py                 1.3.0
virtualenv                   20.17.1
Werkzeug                     2.2.2
wheel                        0.37.1
wrapt                        1.14.1
xgboost                      1.5.2
yarl                         1.8.2
zipp                         3.11.0
```
```

**Comment by anakinxc** [comment]
`cmake: command not found `

build from source 需要一下几个工具
cmake, ninja-build, nasm

同时最好用 gcc11.2

可以参考一下 spu 的[文档](https://github.com/secretflow/spu/blob/main/CONTRIBUTING.md#build) 

元旦快乐

**Comment by anakinxc** [comment]
@zhouaihui 文档这里看起来需要补充一下，先 assign 给你了

**Comment by Stu-Yang** [comment]
@anakinxc 目前已经成功安装，谢谢，祝你新年快乐～
